tests/robustness/checker/checker.go

```diff
@@ -202,8 +202,7 @@ func (chk *Checker) TakeSnapshot(ctx context.Context, sourceDir string) (snapID
 		return snapID, err
 	}
 
-	chk.snapshotMetadataStore.AddToIndex(snapID, allSnapshotsIdxName)
-	chk.snapshotMetadataStore.AddToIndex(snapID, liveSnapshotsIdxName)
+	chk.snapshotMetadataStore.IndexOperation(snapID, map[string]bool{allSnapshotsIdxName: true, liveSnapshotsIdxName: false})
```

Comment:
This doesn't solve the issue. Again, we need to update any index changes within the same lock-unlock block that the metadata Store takes place in, to make it atomic. Prevent this from happening:
me: Store(key)
you: Delete(key)
you: RemoveFromIndex(key, "ActiveKeys")
me: AddToIndex(key, "ActiveKeys")

This is bad because after this series of non-atomic operations we have "key" in the "ActiveKeys" index even though it has been deleted from the metadata store. Two different goroutines were modifying the state of the store and indexes independently. We need it atomic. Either I can change the store and indexes *all at once*, or I have to wait until I can do so (wait on a mutex).